### ISSUE TITLE
Use another genome/annotation?

### ISSUE THREAD

**Issue by dining-philosopher** [openvax/isovar#123]
Hello!

I wonder can i run isovar against some other genome and/or annotation file? Our pipeline uses hg38.analysisset.fa and gencode.v31.chr_patch_hapl_scaff.annotation.gtf, but isovar uses Homo_sapiens.GRCh38.*.fa.gz and Homo_sapiens.GRCh38.102.gtf.gz which it downloads by itself, which results at least in some inconsistencies in gene names.

Is it possible to set genome/annotation from python? It should be obviously so cos the command line isovar accepts genome parameter, but your readme says nothing about this.

Can i read the list of genome names accepted as the mentioned parameter?

Thanks in advance.

**Comment by iskandr** [comment]
Yes. Isovar uses the annotation attached to each variant, so from Python you can load the variants with any PyEnsembl genome and pass them to `run_isovar`:

```python
import pyensembl
import varcode
from isovar import run_isovar

variants = varcode.load_vcf("variants.vcf", genome=pyensembl.EnsemblRelease(93))
results = run_isovar(variants=variants, alignment_file="tumor-rna.bam")
```

For your own annotation, build a `pyensembl.Genome` from its GTF and transcript FASTA files, index it with `genome.index()`, and pass it the same way. Gene and transcript names then come from that annotation, which addresses the name inconsistencies you saw. Align the RNA to the same assembly as the annotation.

I checked this on the b16 test fixtures: a `Genome` built from local GTF/FASTA files gives the same proteins as the matching `EnsemblRelease`.

On the command line, `--genome` accepts an assembly name (`GRCh38`, `hg19`, `GRCm38`, …) and uses the most recent Ensembl release installed for it. Selecting a specific release there is tracked in #122.

This is now documented in the README ([choosing the annotation](https://github.com/openvax/isovar#installation)), released in isovar 1.31.1. Closing; please reopen if something doesn't work for your GENCODE setup.
